swift/swift.go

```diff
@@ -453,16 +455,54 @@ func (o *FsObjectSwift) Open() (in io.ReadCloser, err error) {
 	return
 }
 
+func Min(x, y int64) int64 {
+    if x < y {
+        return x
+    }
+    return y
+}
+
 // Update the object with the contents of the io.Reader, modTime and size
 //
 // The new object may have been created if an error is returned
 func (o *FsObjectSwift) Update(in io.Reader, modTime time.Time, size int64) error {
+	segmentSize, err := strconv.ParseInt(fs.ConfigFile.MustValue(o.swift.Name(), "segment-size"), 10, 64)
```

Comment:
I'd prefer this was a command line option rather than something in the config file - see `--drive-chunk-size` for an example.